python/dlisio/plumbing/channel.py

```diff
@@ -86,16 +86,32 @@ class Channel(BasicObject):
 
     def __init__(self, obj = None, name = None, lf=None):
         super().__init__(obj, name = name, type = 'CHANNEL', lf=lf)
-        # The numpy data type of the sample array
-        self._frame        = None
 
     @property
     def frame(self):
-        if self._frame is not None:
-            return lookup(self.logicalfile, obname('FRAME'), self._frame)
+        if self.logicalfile is None:
+            msg = 'Unable to lookup frame, {} has no logical file'
+            logging.info(msg.format(self))
+            return None
+
+        # Find the frame(s) that are claiming ownership over this channel
+        frames = []
+        for frame in self.logicalfile.frames:
+            for channel in frame.channels:
+                if channel.attic.name != self.attic.name: continue
+                frames.append(frame)
```

Comment:
(Might be good to test on frames containing channels with the same name but different origins. If agreed, let me know; I'll create a file)